src/gui/symbology-ng/qgssymbolslistwidget.h

```diff
@@ -30,7 +30,7 @@ class GUI_EXPORT QgsSymbolsListWidget : public QWidget, private Ui::SymbolsListW
     Q_OBJECT
 
   public:
-    QgsSymbolsListWidget( QgsSymbolV2* symbol, QgsStyleV2* style, QMenu* menu, QWidget* parent );
+    QgsSymbolsListWidget( const QgsVectorLayer * layer, QgsSymbolV2* symbol, QgsStyleV2* style, QMenu* menu, QWidget* parent );
```

Comment:
unfortunately this API break isn't allowed. You'll need to make the layer optional and add checks to qgssymbolslistwidget.cpp so that all references to mLayer are optional. Looks like this only affects the data defined buttons, so those could be either disable or hidden if no layer is specified...